### PR TITLE
[ENH] Add load_classification_train_test wrapper for label encoding

### LINKED ISSUE #2807: [ENH] Add a label encoder option when loading dataset
### Describe the feature or idea you want to propose

When loading datasets, we often get class labels as strings, which is not the best case when wanting to do plotting or other operation (e.g. plotting samples coloured by their class label).

It would be nice to have an option as a boolean parameter to apply (and return) a label encoder directly when loading, instead of having to do it outside this function every time we want too.

### Describe your proposed solution

Add a boolean parameter (default to false to not change default behaviour) that applies and returns a label encoder to the class labels when loading classification datasets.

Some thinking might be needed on how to handle properly the case where we load the training data, and then need to re-use the label encoder for the testing data.

### Describe alternatives you've considered, if relevant

_No response_

### Additional context

I often need to do this when loading classification datasets, so I might not be the only one (hopefuly!)

### PR DESCRIPTION
**Pull request by satwiksps**
## Reference Issue
- Fixes #2807.

## What does this implement/fix? Explain your changes.

This PR implements a new wrapper function, `load_classification_train_test`, in `aeon/datasets/_data_loaders.py`.

As discussed in issue #2807, this function addresses the need to easily load classification datasets with target labels automatically encoded as integers (using `sklearn.LabelEncoder`) without modifying the signature of the base `load_classification` function. It also simplifies the common workflow of loading both train and test splits simultaneously.

**Key changes:**
1.  **New Function:** Added `load_classification_train_test`. 
    * Loads both train and test splits in one call.
    * Includes an `encode_labels` parameter (default `True`) to automatically fit a `LabelEncoder` on the training set and transform both splits.
    * Optionally returns the fitted encoder via `return_encoder=True`.
    * Docstrings match the style and content of the original `load_classification` function for consistency.
2.  **Export:** Exposed the function in `aeon/datasets/__init__.py`.
3.  **Testing:** Added a unit test in `aeon/datasets/tests/test_data_loaders.py` to verify integer conversion, encoder return behavior, and the option to disable encoding.

## Does your contribution introduce a new dependency? If yes, which one?

No

## Any other comments?

No

### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you **after** the PR has been merged.
- [x] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

### For new estimators and functions
- [ ] I've added the estimator/function to the online [API documentation](https://www.aeon-toolkit.org/en/latest/api_reference.html).
- [ ] (OPTIONAL) I've added myself as a `__maintainer__` at the top of relevant files and want to be contacted regarding its maintenance. Unmaintained files may be removed. This is for the full file, and you should not add yourself if you are just making minor changes or do not want to help maintain its contents.

### For developers with write access
- [ ] (OPTIONAL) I've updated aeon's [CODEOWNERS](https://github.com/aeon-toolkit/aeon/blob/main/CODEOWNERS) to receive notifications about future changes to these files.

### CODE REVIEW

**Review by baraline — CHANGES_REQUESTED** [review]
Hi, the code itself is good, but (minor point) I'm unsure about naming. 

We already have a function named load_classification, which expects a split paramater. 

We don't want that paramater here as it would mean we need to carry around the label encoder, so we want to return both train and test, as you did. 

To avoid contusion on the expected paramater and output of the function, I'd rather rename it to something like ` load_classification_train_test`.
The encoded part being optional, including it in the naming looks weird to me.

**Review by MatthewMiddlehurst — CHANGES_REQUESTED** [review]
Yes agreed could use a better name, Antoine's suggestion seems fine if the optional encoding bit is included.

Documentation should be improved and match the original function where possible.

**Comment by MatthewMiddlehurst** [comment]
Could you explain what has changed? by force-pushing you damage the commit history so we cannot easily compare what you had before to what you have now.

**Comment by satwiksps** [comment]
> by force-pushing you damage the commit history

@MatthewMiddlehurst I am sorry for this. I will keep this in mind from now on.

> Could you explain what has changed

I have updated the PR based on the feedback to renamethe function and make the encoding behavior optional.

### Summary of Changes

1.  **Renaming**
    - Renamed the function from `load_classification_encoded` to `load_classification_train_test`.

2.  **Optional Encoding Logic**
    - Added a new parameter `encode_labels=True`.
    - **Default Behavior (`True`):** The function fits a `LabelEncoder` on the training set and transforms both train and test labels to integers.
    - **Optional Behavior (`False`):** Users can pass `encode_labels=False` to get the raw string labels while still benefiting from the convenience of loading both splits simultaneously.

3.  **Improve Documentation**
    - Updated the docstring parameter descriptions to match the original `load_classification` function exactly, ensuring documentation consistency across the API.
    - Added an `Examples` section demonstrating the default usage, the return of the encoder, and the no-encoding case.
    - **Note:** I am still not fully confident about the documentation part. If there are further improvements or conventions I missed, I am more than willing to make changes

4.  **Test Coverage**
    - Updated `test_data_loaders.py` to test the new function name.
    - Added a specific assertion to verify that `encode_labels=False` correctly returns string labels.


Apologies again for the force push. To help you trace the changes, you can refer to:
- **Previous state:** [6fa2bad](https://github.com/aeon-toolkit/aeon/commit/6fa2badd205e68f3511152659fdf32e9a29171cf) (The initial implementation of `load_classification_encoded`)
- **Major changes:** [f25bcab](https://github.com/aeon-toolkit/aeon/commit/f25bcab31bf5ada40a0fe7fd5bc287dd79a398d5) (This has the function renaming, addition of optional encoding logic and most of the doc changes)
- **Minor docstring changes:** [08614f8](https://github.com/aeon-toolkit/aeon/commit/08614f88259011c51b313f142cf529bea79fb4cf) (This has only some minor changes to the docstring of thr returns section)

File: aeon/datasets/_data_loaders.py [hunk 1]
```diff
@@ -1427,6 +1429,114 @@ def load_classification(
...
+
+    Examples
+    --------
+    >>> from aeon.datasets import load_classification_train_test
+    >>> X_train, y_train, X_test, y_test = load_classification_train_test(
+    ...     "ArrowHead"
+    ... ) # doctest: +SKIP
+    """
+    if return_encoder:
+        encode_labels = True
```
Comment (MatthewMiddlehurst) [inline aeon/datasets/_data_loaders.py:1354]: would raise an error instead IMO.

File: aeon/datasets/_data_loaders.py [hunk 2]
```diff
@@ -1427,6 +1429,114 @@ def load_classification(
...
+        split="test",
+        extract_path=extract_path,
+        return_metadata=False,
+        load_equal_length=load_equal_length,
+        load_no_missing=load_no_missing,
+    )
+
+    le = None
+    if encode_labels:
+        le = LabelEncoder()
```
Comment (MatthewMiddlehurst) [inline aeon/datasets/_data_loaders.py:1379]: just create the class in 1 line, no real overhead here.

**Review by MatthewMiddlehurst — COMMENTED** [review]
Please add this to the documentation. 

@baraline any thoughts when you get some free time? I'm fine with it if this solves your issue.

File: aeon/datasets/_data_loaders.py [hunk 3]
```diff
@@ -1427,6 +1429,114 @@ def load_classification(
...
+        this function will look in the extract_path first, and if it is not present,
+        attempt to download the data from www.timeseriesclassification.com, saving it to
+        the extract_path.
+    extract_path : str, default=None
+        the path to look for the data. If no path is provided, the function
+        looks in `aeon/datasets/local_data/`. If a path is given, it can be absolute,
+        e.g. C:/Temp/ or relative, e.g. Temp/ or ./Temp/.
+    return_metadata : boolean, default = False
+        If True, returns a dictionary of metadata found in the header.
+    return_encoder : bool, optional (default=False)
```
Comment (TonyBagnall) [inline aeon/datasets/_data_loaders.py:1310]: we decided the word "optional" was redundant, just

return_encoder : bool, default=False

File: aeon/datasets/_data_loaders.py (hunk 1 above)
Comment (TonyBagnall) [reply aeon/datasets/_data_loaders.py:1354]: agreed

**Review by TonyBagnall — CHANGES_REQUESTED** [review]
looks fine but best wait for the new load_classification.

**Comment by TonyBagnall** [comment]
hi, this needs resolving with data loaders, I will convert to draft for now, feel free to reopen if you want to continue @satwiksps

**Comment by TonyBagnall** [comment]
closing this now, feel free to reopen if you want to continue
